src/sources/generator.rs

```diff
@@ -153,60 +233,105 @@ mod tests {
     }
 
     #[tokio::test]
-    async fn limits_count() {
+    async fn round_robin_limits_count() {
         let mut rx = runit(
-            r#"lines = ["one", "two"]
+            r#"format = "round_robin"
+               lines = ["one", "two"]
                count = 5"#,
         )
         .await;
 
-        for _ in 0..10 {
+        for _ in 0..5 {
             assert!(matches!(rx.poll().unwrap(), Ready(Some(_))));
         }
         assert_eq!(rx.poll().unwrap(), Ready(None));
     }
```

Comment:
This now emits 5 lines instead of 10 - I think this should be a breaking change, let's add `!` to the PR title.